client/blocks/eligibility-warnings/excessive-disk-space.tsx

```diff
@@ -0,0 +1,41 @@
+import { isBusinessPlan, isEcommercePlan, isProPlan } from '@automattic/calypso-products';
+import { localize, LocalizeProps } from 'i18n-calypso';
+import { useSelector } from 'react-redux';
+import PlanStorage from 'calypso/blocks/plan-storage';
+import { canCurrentUser } from 'calypso/state/selectors/can-current-user';
+import { getSiteSlug, getSitePlanSlug } from 'calypso/state/sites/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+
+const ExcessiveDiskSpace = ( { translate }: { translate: LocalizeProps[ 'translate' ] } ) => {
+	const selectedSiteId = useSelector( getSelectedSiteId );
+
+	const siteSlug = useSelector( ( state ) => getSiteSlug( state, selectedSiteId ) );
+	const canUserUpgrade = useSelector( ( state ) =>
+		canCurrentUser( state, selectedSiteId ?? 0, 'manage_options' )
+	);
+	const sitePlanSlug = useSelector( ( state ) => getSitePlanSlug( state, selectedSiteId ) ?? '' );
+
+	const planHasTopStorageSpace =
+		isBusinessPlan( sitePlanSlug ) || isEcommercePlan( sitePlanSlug ) || isProPlan( sitePlanSlug );
```

Comment:
Looks like starting next week, users will be able to upgrade the Pro plan to a Business or eCommerce plan (pdtkmj-c5-p2), so we should remove the `isProPlan` check from here once that's possible.

I guess we can propose that change to https://github.com/Automattic/wp-calypso/pull/64370 (cc @claudiucelfilip).

Reply:
@mmtr Thanks for bringing this up!
I opened #64657 to remove the `isProPlan` for the upcoming plan overhaul revert.